test/common_utils.py

```diff
@@ -25,7 +25,9 @@ def loop(op, in_dims, out_dim, batch_size, *batched_args, **kwarg_values):
         flat_args, args_spec = pytree.tree_flatten(batched_args)
         flat_dims, dims_spec = pytree.tree_flatten(in_dims)
         assert(args_spec == dims_spec)
-        new_args = [a.select(in_dim, idx) if in_dim is not None else a for a, in_dim in zip(flat_args, flat_dims)]
+        # Tensors in new_args could be required to be contiguous for certain ops like cdist and cdist_backward
+        # Thus, a.select(...) -> a.select(...).contiguous()
+        new_args = [a.select(in_dim, idx).contiguous() if in_dim is not None else a for a, in_dim in zip(flat_args, flat_dims)]
```

Comment:
Do we actually need this change?
- If cdist does not support non-contiguous inputs, then that's a bug.
- cdist_backward should not get directly invoked by `loop` because it only exists in aten.

Reply:
Without this patch, expected results can't be computed for certain batched inputs due to the same issue as here: https://github.com/pytorch/functorch/pull/306/files/04ae2b1dbd3140b389a684c75c1cc945f24e2180#r769666360

cdist supports non-contig inputs but for forward pass only.